api4/channel.go

```diff
@@ -486,6 +487,28 @@ func viewChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func getChannelsByIds(c *Context, w http.ResponseWriter, r *http.Request) {
+	channelIds := model.ArrayFromJson(r.Body)
+	if len(channelIds) == 0 {
+		c.SetInvalidParam("channel_ids")
+		return
+	}
+
+	for _, cid := range channelIds {
+		if len(cid) != 26 {
+			c.SetInvalidParam("channel_id")
+			return
+		}
+	}
+
+	if channels, err := app.GetChannelsByIds(channelIds, c.Session.UserId); err != nil {
```

Comment:
Hmmm, I think we need to detach this from a user id. I'm thinking it might better to make this for a team and only for open/public channels.

So the endpoint would be `/teams/{team_id}/channels/ids` and it would only have to check the "view_team" permission and wouldn't need a user id.

Does that make sense?

Apologies that I didn't fully think this out before I put it in the spreadsheet